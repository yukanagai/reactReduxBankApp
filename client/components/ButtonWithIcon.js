import React from 'react';

// CREATES A BUTTON ELEMENT 
//PASS IN TEXT, ICON, OR BOTH

const ButtonWithIcon = (props) => {
	const text = (props.buttonText) ? props.buttonText : null;
	const icon = (props.iconClass) ? <i className={props.iconClass}></i> : null;
	
	return (
		<button id={props.id} className='ui button' onClick={props.onClickAction}>
			{icon}
			{text}
		</button>
	);
};

export default ButtonWithIcon;