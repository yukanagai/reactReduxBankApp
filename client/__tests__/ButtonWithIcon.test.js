import React from 'react';
import renderer from 'react-test-renderer';
import ButtonWithIcon from '../components/ButtonWithIcon.js';

// Snapshot for WithdrawModal React Component
test('BUTTON WITH ICON --- Snapshot',()=>{
    const tree = renderer.create(
    	<ButtonWithIcon	
    		text='sample btn text'
    		iconClass='minus icon'/>
    ).toJSON();
   	expect(tree).toMatchSnapshot();
});