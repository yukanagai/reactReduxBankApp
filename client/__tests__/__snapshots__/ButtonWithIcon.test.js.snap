// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BUTTON WITH ICON --- Snapshot 1`] = `
<button
  className="ui button"
  id={undefined}
  onClick={undefined}
>
  <i
    className="minus icon"
  />
</button>
`;
